journalsmanager/utils.py

```diff
@@ -145,25 +146,62 @@ def export_issns(rows):
 def export_to_autocomplete(rows):
     data = []
     try:
+        canonicalBibs = []
+        bibcodeFile = config.get('CANONICAL_BIBS', None)
+        if bibcodeFile:
+            bibcodeFile = JDB_DATA_DIR + '/' + bibcodeFile
+            with open(bibcodeFile, 'r') as fb:
+                for l in fb.readlines():
+                    canonicalBibs.append(l.strip())
+        cites = {}
+        citationFile = config.get('CITATION_COUNTS', None)
+        if citationFile:
+            citationFile = JDB_DATA_DIR + '/' + citationFile
+            with open(citationFile, 'r') as fc:
+                for l in fc.readlines():
+                    (bibcode, tcit, rcit) = l.strip().split('\t')
+                    if cites.get(bibcode, None):
+                        cites[bibcode] += (int(tcit) + int(rcit))
+                    else:
+                        cites[bibcode] = (int(tcit) + int(rcit))
         for r in rows:
             bibstem = r.get('bibstem', None)
+            if len(bibstem) < 5:
+                bibstem = bibstem.ljust(5, '.')
             names = list()
-            if r.get('name', None):
-                names.append(r.get('name', None))
-            if r.get('translated_name', None):
-                names.append(r.get('translated_name', None))
-            if r.get('native_name', None):
-                names.append(r.get('native_name', None))
-            if r.get('transliterated_name', None):
-                names.append(r.get('transliterated_name', None))
-            if bibstem and names:
-                data.append({'value': bibstem, 'label': names})
-            elif not bibstem:
-                print('what the hell? %s' % str(r))
-        result = {'data': data}
+            bibcodeList = [c for c in canonicalBibs if bibstem in c]
+            bibcodeCount = len(bibcodeList)
+            if bibcodeCount > 0:
+                citeSum = sum(cites.get(x, 0) for x in bibcodeList)
+                rank = bibcodeCount + citeSum
+                if r.get('name', None):
+                    names.append(r.get('name', None))
```

Comment:
I debated that with myself actually.  I opted to be consistent mainly so that you see the same value in code being assigned.  I don't know if using dict.get causes lots of extra overhead, but it might -- it's basically an if-then statement.